### PR TITLE
Update the diagnostics rolling perf tests

### PR DESCRIPTION
**Pull request by mavasani**
1. Split out the tests into ones with built-in IDE analyzers and ones with only third party analyzers
2. Tests depends on closed side changes (add/update VS tao actions)

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @basoundr @heejaechang @KevinH-MS @dotnet/roslyn-analysis for review

**Comment by mavasani** [comment]
FYI: I'll be working with @basoundr to port these tests to the new perf infrastructure. This change is merely updating the old perf tests (which are currently not executing)

File: src/EditorFeatures/CSharpTest/PerfTests/CSharpPerfFullSolutionDiagnosticsThirdPartyAnalyzers.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  <CleanupScenario>
+    <CloseFile SaveFile="false"/>
+  </CleanupScenario>
+
+  <CleanupTest>
+    <!-- Revert the ruleset changes -->
+    <ReplaceRuleset SourceRulesetFile="Tools\Targets\Roslyn_Default.ruleset" DestinationRulesetFile="Tools\Targets\Roslyn.ruleset"/>
+    <!-- Remove the added analyzers -->
+    <RemoveAnalyzersFromSolution CSharpAnalyzers="References\Analyzers\StyleCop.Analyzers.1.0.0\analyzers\dotnet\cs\StyleCop.Analyzers.dll;References\Analyzers\StyleCop.Analyzers.1.0.0\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll;References\Analyzers\StyleCop.Analyzers.1.0.0\analyzers\dotnet\cs\Newtonsoft.Json.dll"/>
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/PerfTests/CSharpPerfFullSolutionDiagnosticsThirdPartyAnalyzers.xml]: is this needed, you are not saving it to disk anyway

File: src/EditorFeatures/CSharpTest/PerfTests/CSharpPerfGoldilocksTypingFullSolutionDiagnosticsThirdPartyAnalyzers.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  <CleanupScenario>
+    <CloseFile SaveFile="false"/>
+  </CleanupScenario>
+
+  <CleanupTest>
+    <!-- Revert the ruleset changes -->
+    <ReplaceRuleset SourceRulesetFile="Tools\Targets\Roslyn_Default.ruleset" DestinationRulesetFile="Tools\Targets\Roslyn.ruleset"/>
+    <!-- Remove the added analyzers -->
+    <RemoveAnalyzersFromSolution CSharpAnalyzers="References\Analyzers\StyleCop.Analyzers.1.0.0\analyzers\dotnet\cs\StyleCop.Analyzers.dll;References\Analyzers\StyleCop.Analyzers.1.0.0\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll;References\Analyzers\StyleCop.Analyzers.1.0.0\analyzers\dotnet\cs\Newtonsoft.Json.dll"/>
```
Comment (heejaechang) [inline src/EditorFeatures/CSharpTest/PerfTests/CSharpPerfGoldilocksTypingFullSolutionDiagnosticsThirdPartyAnalyzers.xml]: same for all these except the one that saves proj files

**Comment by heejaechang** [comment]
:+1:

File: src/EditorFeatures/CSharpTest/PerfTests/CSharpPerfGoldilocksTypingFullSolutionDiagnosticsThirdPartyAnalyzers.xml (hunk 2 above)
Comment (mavasani) [inline src/EditorFeatures/CSharpTest/PerfTests/CSharpPerfGoldilocksTypingFullSolutionDiagnosticsThirdPartyAnalyzers.xml]: Sure, will address this when we port these tests to the new location.

**Comment by mavasani** [comment]
retest this please

**Pull request merged**